apps/site/assets/ts/stop/components/StopPageRedesign.tsx

```diff
@@ -8,30 +8,68 @@ import StopPageHeaderRedesign from "./StopPageHeaderRedesign";
 import Loading from "../../components/Loading";
 import StopPageDepartures from "./StopPageDepartures";
 import Alerts from "../../components/Alerts";
-import { Route } from "../../__v3api";
+import { Alert, Route } from "../../__v3api";
 import { useSchedulesByStop } from "../../hooks/useSchedules";
-import { useAlertsByStop } from "../../hooks/useAlerts";
 import StopPageOverlay from "./StopPageOverlay";
+import { useAlertsByRoute, useAlertsByStop } from "../../hooks/useAlerts";
+import { isCurrentAlert, routeWideAlerts } from "../../models/alert";
 
 const StopPageRedesign = ({
   stopId
 }: {
   stopId: string;
 }): ReactElement<HTMLElement> => {
+  const [routeIdState, setRouteIdState] = useState<string[]>([]);
+  const [routeState, setRouteState] = useState<Route[]>([]);
```

Comment:
question: Are these needed as separate pieces of state? Couldn't routeIds be calculated from routeState?

Reply:
I did some re-arranging of code, and managed to remove the state requirements